contracts/SimpleProviders/TimedDealProvider/TimedDealProvider.sol

```diff
@@ -20,12 +21,23 @@ contract TimedDealProvider is LockDealState, DealProviderState, BasicProvider {
         lockDealNFT = _lockDealNFT;
         name = "TimedDealProvider";
     }
-    
+
     function _withdraw(
         uint256 poolId,
         uint256 amount
     ) internal override firewallProtectedSig(0x9e2bf22c) returns (uint256 withdrawnAmount, bool isFinal) {
         (withdrawnAmount, isFinal) = provider.withdraw(poolId, amount);
```

Comment:
we need to skip this line.
on current version we make :
old.amount -= amount
new.amount = old.amount
old.amount = 0(missing)

lets set the new directly to old.amount - amount
and the old to 0

the if on line 31 can be simple